wandb/sdk/internal/internal_api.py

```diff
@@ -770,22 +774,107 @@ def upsert_project(self, project, id=None, description=None, entity=None):
         return response["upsertModel"]["model"]
 
     @normalize_exceptions
-    def pop_from_run_queue(self, entity=None, project=None):
+    def get_project_run_queues(self, entity, project):
+        query = gql(
+            """
+        query Project($entity: String!, $projectName: String!){
+            project(entityName: $entity, name: $projectName) {
+                runQueues {
+                    id
+                    name
+                    createdBy
+                    access
+                }
+            }
+        }
+        """
+        )
+        variable_values = {
+            "projectName": project,
+            "entity": entity,
+        }
+
+        res = self.gql(query, variable_values)
+        if res.get("project") is None:
+            raise Exception(
+                "Error fetching run queues for {}/{} check that you have access to this entity and project".format(
+                    entity, project
+                )
+            )
+
+        return res["project"]["runQueues"]
+
+    @normalize_exceptions
+    def push_to_run_queue(self, queue_name, launch_spec):
+        # TODO(kdg): add pushToRunQueueByName to avoid this extra query
+        queues_found = self.get_project_run_queues(
```

Comment:
We should likely just make this idempotent.  What's the case where this method is called but the queue doesn't exist currently?

Reply:
This could happen if the user uses launch_add, and specifies a queue name that doesn't exist. Looking at this now I think that we should make the case where the queue is "default" create the queue then add the item. otherwise, just log to the terminal that the queue does not exist